src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,105 @@
+import './NewCommentForm.scss';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../../types';
+
+type Props = {
+  commentsList: Comment[] | undefined;
+  setCommentsList: (arg:Comment[]) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+                                                  commentsList, setCommentsList,
+                                                }) => {
+  const [yourname, setYourname] = useState('');
+  const [youremail, setYouremail] = useState('');
+  const [yourcomment, setYourcomment] = useState('');
+  const [error, setError] = useState(false);
+
+  const addComment = (newComment: Comment) => {
+    if (commentsList) {
+      commentsList.unshift(newComment);
+      const copy = [...commentsList];
+
+      setCommentsList(copy);
+    }
+  };
+
+  const submitHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (yourname && youremail && yourcomment && commentsList) {
+      const newComment = {
+        id: commentsList.length + 1,
```

Comment:
Here, you are using the length of the array as the item's ID. I recommend using an external library to generate unique codes instead of using hardcoded values directly in the code. I suggest using the uuid library. This will improve the flexibility and scalability of the project.